test/e2e/tinkerbellhardwarecount.go

```diff
@@ -0,0 +1,43 @@
+package e2e
+
+import (
+	_ "embed"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed tinkerbell_hardware_count.yaml
+var tinkerbellHardwareCountFile string
+
+// TinkerbellTest maps each Tinkbell test with the hardware count needed for the test.
+type TinkerbellTest struct {
+	Name  string `yaml:"name"`
+	Count int    `yaml:"count"`
+}
+
+// GetTinkerbellHardwareMap returns a map of tink tests with associated hardware count.
+func GetTinkerbellHardwareMap() (map[string]int, error) {
+	tinkerbellHardwareMap, err := readTinkerbellTestFile()
+	if err != nil {
+		return nil, fmt.Errorf("error getting hardware count for tinkerbell tests: %v", err)
+	}
+	return tinkerbellHardwareMap, nil
+}
+
+func readTinkerbellTestFile() (map[string]int, error) {
+	tinkerbellTests := make(map[string][]TinkerbellTest)
+	err := yaml.Unmarshal([]byte(tinkerbellHardwareCountFile), &tinkerbellTests)
+	if err != nil {
+		return nil, fmt.Errorf("unable to Unmarshal %s file: %v", TinkerbellHardwareCountFile, err)
+	}
+	tinkerbellTestsHardwareCount, ok := tinkerbellTests[tinkerbellTestsHardwareCountIdentifier]
+	if !ok {
+		return nil, fmt.Errorf("error in type assertion: %v", tinkerbellTests[tinkerbellTestsHardwareCountIdentifier])
+	}
```

Comment:
I think we can format the YAML a little differently so the identifier isn't necessary.

If we want something really simple we could use `map[string]int`. Tests require unique names so we should never have duplicate map keys.

```
TestTinkerbellKubernetes123AWSIamAuth: 2
TestTinkerbellKubernetes127AWSIamAuth: 1
...
```

If we want to extensibility per test then `[]map[string]string`.

```
- name: TestTinkerbellKubernetes123AWSIamAuth
  count: 2
- name: TestTinkerbellKubernetes127AWSIamAuth
  count: 1
...
```

Reply:
Yes, I did think about this. I went with this approach at first. But to have some conformance with the existing Yaml like `skipped_tests.yaml`, I decided to add an Identifier in place just to make the yaml more readable.

Reply:
I don't think we need to follow suit with `skipped_tests.yaml`; the configurations serve different purposes.